### PR TITLE
Add citing works page and gh actions to update

### PR DESCRIPTION
**Pull request by sneakers-the-rat**
OK so here's da deal, the page reads from the zotero api into `_data/citing_papers/items.json` from the github action that runs every day at midnight. if there are changes it will make a pull request for approval.

anyway it's like this:

https://github.com/user-attachments/assets/3315554f-9e26-45cf-8468-3eed40900db4

filter by any combo of tags or search freeform text from any field. no sorting but cmon lol

to make this work had to move to the new style of github pages build using github actions, so if this looks good i'll test it and make sure everything works after we merge, since there is some weirdness about being able to run github actions from a branch before the action exists on the default branch

--- 

for anyone else reading this, to use this it's just a matter of
- make a zotero collection
- add tags to stuff
- update the URL parameters so helpfully hardcoded over here to wherever your location is in whatever org/account: https://github.com/Aharoni-Lab/Aharoni-Lab.github.io/blob/ad704c4bb5ea4e1b6cdb6f14d6fe6a18c16e316f/.github/workflows/update_citing_papers.yaml#L22
- make a zotero API key that has read access to that collection/org/user and put it in your `ZOTERO_READONLY` secret for the repo
- the rest is just jekyll stuff: add the partials, css, js and then use `{% include papers=site.data.wherever_you_put_it %}` in a page

Note that the github actions part has NOT been tested yet, but we'll get it working as soon as my PI has a chance to review this and marge it, and any updates I do will be on top of the https://github.com/Aharoni-Lab/Aharoni-Lab.github.io/tree/citing-papers branch

### CODE REVIEW

File: .github/workflows/update_citing_papers.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      # your collection's ID
+      COLLECTION_KEY: "KRCY8EJ7"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Check if update
+        run: |
+          LAST_VERSION=$(<last_modified_version)
+          curl https://api.zotero.org/groups/${{ env.ORG_KEY }}/collections/${{ env.COLLECTION_KEY }}/items/top\?limit\=1 -H "Authorization: Bearer ${{ secrets.ZOTERO_READONLY }}" -H "If-Modified-Since-Version: ${LAST_VERSION}" -D headers.txt
```
Comment (sneakers-the-rat) [inline .github/workflows/update_citing_papers.yaml:28]: whats da deal with these lines

File: Gemfile [hunk 2]
```diff
@@ -1,11 +1,9 @@
 source 'https://rubygems.org'
-gem 'github-pages', group: :jekyll_plugins
 
-gem "jekyll-remote-theme"
+gem "jekyll", "~> 4.3"
```
Comment (sneakers-the-rat) [inline Gemfile:3]: how come u change this

File: Publications/index.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+---
+title: "Publications"
+---
```
Comment (sneakers-the-rat) [inline Publications/index.md:3]:
```suggestion
---
title: "Papers"
---
```

This should be named "papers" instead plz

**Review by sneakers-the-rat — COMMENTED** [review]
This is my self-review on my own PR as a demo for https://wiki.aharoni-lab.com/Code_Review

File: .github/workflows/update_citing_papers.yaml (hunk 1 above)
Comment (sneakers-the-rat) [reply .github/workflows/update_citing_papers.yaml:28]: Here is the answer to ur question, resolved

File: Publications/index.md (hunk 3 above)
Comment (sneakers-the-rat) [reply Publications/index.md:3]: no actually

**Review by daharoni — APPROVED** [review]
Looks great!

**Pull request merged**